app/Config/Modules.php

```diff
@@ -31,6 +31,29 @@ class Modules extends BaseModules
      */
     public $discoverInComposer = true;
 
+    /**
+     * The Composer package list for Auto-Discovery
+     * This setting is optional.
+     *
+     * E.g.:
+     *   [
+     *       'only' => [
+     *           // List up all packages to auto-discover
+     *           'codeigniter4/sheild',
+     *       ],
+     *   ]
+     *   or
+     *   [
+     *       'exclude' => [
+     *           // List up packages to exclude.
+     *           'pestphp/pest',
+     *       ],
+     *   ]
+     *
+     * @var array
```

Comment:
An alternate suggestion to consider:
```suggestion
     * @var array<string, bool> Array of "package/name" => "whether to load/exclude"
```
It simplifies the Config but would take more explanation in documentation.

Reply:
How do we handle unlisted packages?

Reply:
However they are handled now. Basically this would be the equivalent of:
```php
$only    = [];
$exclude = [];
foreach ($composerPackages as $package => $flag) {
    if ($flag) {
        $only[] = $package;
    } else {
        $exclude[] = $package;
    }
}
```
Same logic with a simpler Config. I'm not set on this though, just a suggestion for how to simplify the Config structure (but it has its own pitfalls).

Reply:
It is simpler as the array structure, but tihe config seems more difficult to understand for humans.

Reply:
Leave it; proceed.